### PR TITLE
Update Vet_clinic

### PR DESCRIPTION
**Pull request by mumbereh**
### **In this mileston I have implemented the following**
Use database transactions.
Modify and delete data in SQL.
Prepare complex queries that answer analytical questions.

### **Below are the screenshots for my implementation**

i2
2i

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -7,4 +7,33 @@ SELECT date_of_birth FROM Animals WHERE name IN ('Agumon', 'Pikachu');
...
+WHERE date_of_birth > '2022-01-01';
+SAVEPOINT weight_update;
+UPDATE Animals
+SET weight_kg = weight_kg * -1;
+ROLLBACK TO weight_update;
+UPDATE Animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+COMMIT;
+
```
Comment (Shedrack-Sunday) [inline queries.sql]: - [ ] You missed a large portion of the query task asked. I've screenshotted it below to show you whats missing. Kindly carry it out, and don't forget add all screenshots to the PR you generated 👍 

### MISSED QUERY:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @mumbereh          👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the SQL  features ✔️ 
- Nice use of query parameters✔️ 

# Required Changes  🌵  

- [x] You've done well, However, kindly follow the suggestion on the right query pattern properly so your impressive work will have a better implementation 👍 


_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: queries.sql [hunk 2]
```diff
@@ -7,4 +7,33 @@ SELECT date_of_birth FROM Animals WHERE name IN ('Agumon', 'Pikachu');
...
+DELETE FROM Animals
+WHERE date_of_birth > '2022-01-01';
+SAVEPOINT weight_update;
+UPDATE Animals
+SET weight_kg = weight_kg * -1;
+ROLLBACK TO weight_update;
+UPDATE Animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+COMMIT;
```
Comment (ghost) [inline queries.sql]: - [ ] As requested earlier, kindly add this part of the transaction to your `queries.sql` file as it is part of the project requirements.

image

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi @mumbereh , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

YesYouAreGIF (2)




## Highlights

✅ Schema file meets the requiremets 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql [hunk 3]
```diff
@@ -6,5 +6,35 @@ SELECT name FROM Animals WHERE neutered = TRUE AND escape_attempts < 3;
...
 SELECT name, escape_attempts FROM Animals WHERE weight_kg > 10.5;
 SELECT * FROM Animals WHERE neutered = TRUE;
-SELECT * FROM Animals WHERE name<> 'Gabumon';
-SELECT * FROM Animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM Animals WHERE name <> 'Gabumon';
+SELECT * FROM Animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+BEGIN;
+
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
```
Comment (Davidosky007) [inline queries.sql:13]: * Some required transactions are still missing below are the missing transactions 👇 
```
Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
Inside a transaction:
Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
Verify that changes were made.
Commit the transaction.
Verify that changes persist after commit.
Take a screenshot of the results of your actions.
Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)
Take a screenshot of the results of your actions.
```
* This is an example transaction query below 👇 UPDATE the animals table by setting the species column to unspecified, then roll back the change (**for this step you are explicitly asked to verify that change was made and rollbacked, so make sure that they included all steps as in the example below **). **Query** :     
```
BEGIN; -- start transaction

   UPDATE animals
   SET species = 'unspecified'; -- make change

   SELECT species from animals; -- verify that change was made
   ROLLBACK;

   SELECT species from animals; -- verify that change was undone

```
* Kindly use the Hint provided above to write the missing queries

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @mumbereh 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Data modified and deleted properly 👍 
* Well-written PR 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by binhussen — APPROVED** [review]
Hi @mumbereh , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Added Transaction ✔️
- Well Structured Query  ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**